### PR TITLE
Fix "reference to local binding" after fixes for clang-17

### PR DESCRIPTION
**Pull request by azat**
Follow-up for: #49851 (cc @alexey-milovidov)

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Coordination/KeeperStorage.cpp [hunk 1]
```diff
@@ -363,7 +363,7 @@ void KeeperStorage::UncommittedState::applyDelta(const Delta & delta)
             {
                 assert(my_node);
                 my_node->invalidateDigestCache();
-                operation.update_fn(*node);
+                operation.update_fn(*my_node);
```
Comment (antonio2368) [inline src/Coordination/KeeperStorage.cpp:366]: we can simply remove explicit reference bindings and leave `[&]`?

https://godbolt.org/z/W6h14jeeE

Comment (azat) [reply src/Coordination/KeeperStorage.cpp:366]: Yep, but I guess the reason was to explicitly show what should be binded.

**Comment by robot-ch-test-poll3** [comment]
This is an automated comment for commit 749b58a2412c40361b0daf0f293cabefe341c09b with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/49874/749b58a2412c40361b0daf0f293cabefe341c09b/ci_running.html)
The overall status of the commit is 🟢 success

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>Style Check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td>🟢 success</td></tr>
</table>

File: src/Coordination/KeeperStorage.cpp (hunk 1 above)
Comment (azat) [reply src/Coordination/KeeperStorage.cpp:366]: So what would you prefer, simply `auto & [node, acls, last_applied_zxid] = nodes.at(delta.path)` inside lambda?

Comment (antonio2368) [reply src/Coordination/KeeperStorage.cpp:366]: yeah

**Review by antonio2368 — APPROVED** [review]

**Pull request merged**